simuleval/data/dataloader/s2t_dataloader.py

```diff
@@ -75,40 +81,44 @@ def get_source_audio_path(self, index: int):
 
     @classmethod
     def from_files(
-        cls, source: Union[Path, str], target: Union[Path, str]
+        cls, source: Union[Path, str], target: Union[Path, str], tgt_lang: str
     ) -> SpeechToTextDataloader:
         with open(source) as f:
             source_list = [line.strip() for line in f]
         with open(target) as f:
             target_list = [line.strip() for line in f]
-        dataloader = cls(source_list, target_list)
+        with open(tgt_lang, "r") as f:
+            tgt_lang = f.read()
```

Comment:
The source and target files can have a several samples/instances hence the list. We want to be able to set the tag_lang for each of those instances independently, so better to make the tgt_lang a list similar to source/target.

Reply:
Noted

Reply:
This is what s2t_dataloader.py will look like:

```
      with open(source) as f:
            source_list = [line.strip() for line in f]
        with open(target) as f:
            target_list = [line.strip() for line in f]
        with open(tgt_lang, "r") as f:
            tgt_lang_list = [line.strip() for line in f]
        dataloader = cls(source_list, target_list, tgt_lang_list)
        return dataloader
```

Could something like this work:
```
def from_files(
    cls, source: Union[Path, str], target: Union[Path, str], tgt_langs: List[str]
) -> SpeechToTextDataloader:
    with open(source) as f:
        source_list = [line.strip() for line in f]
    with open(target) as f:
        target_list = [line.strip() for line in f]

    if len(source_list) != len(target_list) or len(source_list) != len(tgt_langs):
        raise ValueError("Number of instances in source, target, and tgt_langs must match.")

    dataloader_instances = []
    for source_instance, target_instance, tgt_lang in zip(source_list, target_list, tgt_langs):
        dataloader_instances.append(cls([source_instance], [target_instance], tgt_lang))

    return dataloader_instances
```

Reply:
Yes. See how the source and target list in the dataloader is being handled in the `instance` class, we want something similar for the tgt_lang.
https://github.com/facebookresearch/SimulEval/blob/main/simuleval/evaluator/instance.py#L46-L48

Also see the logic for getting a particular index from the list in the dataloader : https://github.com/facebookresearch/SimulEval/blob/main/simuleval/data/dataloader/dataloader.py#L55-L56